src/saveload/linkgraph_sl.cpp

```diff
@@ -53,32 +53,36 @@ const SaveLoad *GetLinkGraphJobDesc()
 	static std::vector<SaveLoad> saveloads;
 	static const char *prefix = "linkgraph.";
 
+	static const SaveLoad job_desc[] = {
+		SLE_VAR(LinkGraphJob, join_date,        SLE_INT32),
+		SLE_VAR(LinkGraphJob, link_graph.index, SLE_UINT16),
+		SLE_END()
+	};
+
+	/* The member offset arithmetic below is only valid if the types in question
+	 * are standard layout types. Otherwise, it would be undefined behaviour. */
+	static_assert(std::is_standard_layout<LinkGraphSettings>::value, "LinkGraphSettings needs to be a standard layout type");
+
+	/* We store the offset of each member of the #LinkGraphSettings in the
+	 * extra data of the saveload struct. Use it together with the address
+	 * of the settings struct inside the job to find the final memory address. */
+	static SaveLoadAddrProc * const proc = [](void *b, size_t extra) -> void * { return  const_cast<void *>(static_cast<const void *>(reinterpret_cast<const char *>(std::addressof(static_cast<LinkGraphJob *>(b)->settings)) + extra)); };
```

Comment:
I sometimes hate my OCD (I don't have OCD :P), as it leads to these comments that are not wrong, but .. I often wonder if it is worth mentioning, but .. my OCD triggered, so here it goes:

double spaces between `return` and `const_cast` :P